connector/src/main/java/org/geysermc/connector/network/session/cache/WorldCache.java

```diff
@@ -66,13 +78,17 @@ public int increaseAndGetScoreboardPacketsPerSecond() {
     }
 
     /**
-     * Tell the client to hide or show the coordinates
+     * Tell the client to hide or show the coordinates.
+     *
+     * If {@link #isPrefersShowCoordinates()} is true, coordinates will be shown, unless either of the following conditions apply:
+     *
+     * <li> {@link GeyserSession#isReducedDebugInfo()} is enabled
+     * <li> {@link GeyserConfiguration#isShowCoordinates()} is disabled.
      *
-     * @param value True to show, false to hide
      */
-    public void setShowCoordinates(boolean value) {
-        boolean allowUserSetting = !session.isReducedDebugInfo() && session.getConnector().getConfig().isShowCoordinates();
-        showCoordinates = allowUserSetting && value;
-        session.sendGameRule("showcoordinates", allowUserSetting && value);
+    public void setShowCoordinates() {
```

Comment:
This should be renamed to `updateShowCoordinates()` as that makes more sense since it no longer takes a value.